pkg/render/common/data/secrets.go

```diff
@@ -0,0 +1,92 @@
+package data
```

Comment:
I'm not a fan of `common/data`, data to me suggests that it just contains some common "data".
I don't feel like the Secret/TLS creation and resource copying fits, as "data". If we want to keep everything under data in one package it feels like a "render/utils" would be most appropriate for what is in "data" currently. I think if we can't come up with a better package name then I think "utils" is more descriptive than "data".
Saying that I think perhaps we should have a `render/common/secret` package and then any secret manipulation (including CreateTLSSecret) can go in there.

Reply:
I agree, I'm not the hugest fan of data either, I just don't like "utils" either. Maybe we could just have a secrets package, but I'm unsure about adding a package for everything else that's in here (some constants not secret related, config map stuff, pod security).

What do you think, should I just rename it to utils or create a bunch of separate packages that may not have much in them? I'm a little torn...

Reply:
I just saw your comment below being in favour of multiple packages. I'll change that.